vnext/Shared/Utils.h

```diff
@@ -20,6 +20,6 @@ struct Url {
   std::string Target();
 };
 
-std::future<std::string> getApplicationDataPath(const wchar_t *childfolder);
+concurrency::task<std::string> getApplicationDataPath(const wchar_t *childfolder);
```

Comment:
I have checked our internal C++ alias and they do not recommend using `std::future`, ConcRT, or PPL (std::future is based on PPL).
I guess it was a reason why Raymond Chen created a new library that works well with C++/WinRT.
https://github.com/microsoft/wil/blob/master/include/wil/coroutine.h
This way we would need to add dependency on WIL, which we were avoiding so far.

Anyway, the change is good for now. We must follow up on that after the PR is merged.

Reply:
We have few enough cases that we could also likely convert these to WinRT types and just use IAsyncOperation. It's also not entirely ideal, but would avoid concurrency runtime usage. Let me know what you think of that.

Reply:
IMHO, in the current situation it would be the best to use `IAsyncOperation`, but I do not know how difficult to apply it. I will leave it up to your judgement.